Compile fails - undefined reference to `esphome::nextion::Nextion::upload_tft()

### The problem

Compile fails with following error message

```
.pioenvs/weatherstation-002/src/main.cpp.o:(.literal._ZNSt17_Function_handlerIFvvEZ5setupvEUlvE0_E9_M_invokeERKSt9_Any_data+0x4): undefined reference to `esphome::nextion::Nextion::upload_tft()'
.pioenvs/weatherstation-002/src/main.cpp.o: In function `std::_Function_handler<void (), setup()::{lambda()#2}>::_M_invoke(std::_Any_data const&)':
/config/build/weatherstation-002/src/esphome/core/automation.h:167: undefined reference to `esphome::nextion::Nextion::upload_tft()'

```

### Which version of ESPHome has the issue?

latest dev

### What type of installation are you using?

Docker

### Which version of Home Assistant has the issue?

docker image 7ae4d7600985

### What platform are you using?

ESP32

### Board

es[32dev

### Component causing the issue

nextion

### Example YAML snippet

```yaml
display:
    - platform: nextion
      id: nextion0
      uart_id: uart_nextion0
      tft_url: 'http://192.168.0.73/nextiontft/WeatherStation-Small.tft'
      update_interval: 15s
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/esphome/components/nextion/nextion_upload.cpp
+++ b/esphome/components/nextion/nextion_upload.cpp
@@ -1,6 +1,7 @@
+#include "nextion.h"
+
 #ifdef USE_NEXTION_TFT_UPLOAD
 
-#include "nextion.h"
 #include "esphome/core/application.h"
 #include "esphome/core/macros.h"
 #include "esphome/core/util.h"

